src/FSharp.Core/array.fs

```diff
@@ -1936,6 +1936,212 @@ module Array =
 
             result
 
+    [<CompiledName("RandomShuffleWith")>]
+    let randomShuffleWith (random: Random) (source: 'T array) : 'T array =
+        checkNonNull "random" random
+        checkNonNull "source" source
```

Comment:
I think it's time we should use nameof everywhere. Feel free not to change it in this pr, was just a remark

Reply:
Right, it would make more sense for me to do it in bulk in the whole core library (or even in all repository)

Reply:
Well, if we ever figure out what we're doing with #11976...

https://github.com/dotnet/fsharp/assets/14795984/48a06fea-2b5f-4819-b55b-47a5ccd4597a